libr/debug/p/native/linux/linux_debug.c

```diff
@@ -193,52 +206,193 @@ bool linux_set_options (RDebug *dbg, int pid) {
 	return true;
 }
 
-int linux_attach (RDebug *dbg, int pid) {
-	linux_set_options (dbg, pid);
-	int ret = ptrace (PTRACE_ATTACH, pid, NULL, NULL);
+// Local copy for us. Here we keep all traced threads (including the main_pid). 
+static RList *list_traced_threads = NULL;
+
+static int match_pid (const void *pid_o, const void *th_o) {
+	int pid = *(int *)pid_o;
+	RDebug *th = (RDebug *)th_o; 
+	return pid == th->pid;
+}
+
+static void remove_list_thread (void) {
```

Comment:
Within `remove_list_thread` just use `r_list_free` and set the free function in the list when creating it with `r_list_newf`